### DISCUSSION TITLE
Passing Array of Strings from C++

### DISCUSSION THREAD

**Discussion by gammaEmitter** [slint-ui/slint#2324] [Q&A] [answered]
Hello Community!

I hope this is the right way to find help for my issue.

My goal is to populate a ComboBox with Items I hold in an array of strings. This is represented by slint as an
`std::shared_ptr<slint::Model<slint::SharedString>>`

I manage to create the object with std::make_shared and then try to add to the Model by its method
`set_row_data(index_of_element,slint::SharedString(element))`

I keep getting errors of a static_cast when creating the shared_ptr. When I create the pointer in another way, I get an error about using an abstract base class.

What is the usage of the Model wrapper for arrays and how can I solve my issue?

Thanks :)

**Comment by ogoffart** [comment, marked as answer]
Could you paste the exact error message you're getting as well as the relevant lines of code?

Are you using slint::VectorModel ?

Maybe you need to use [std::static_pointer_cast](https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast) ?

**Reply by gammaEmitter** [reply]
Thank you for your quick reply! My issue was resolved.

The error occured when I created the pointer by
`auto model = std::make_shared<slint::Model<slint::SharedString>>();`

and went away when I correctly applied `slint::VecotrModel` instead of the Base Class `slint::Model`
`auto model = std::make_shared<slint::VectorModel<slint::SharedString>>();`

For completion sake, this was the error from the first line

> error: static_cast from 'std::__compressed_pair<std::allocator<slint::Model<slint::SharedString>>, slint::Model<slint::SharedString>> *' to 'std::__compressed_pair<std::allocator<slint::Model<slint::SharedString>>, slint::Model<slint::SharedString>>::_Base2 *' (aka '__compressed_pair_elem<slint::Model<slint::SharedString>, 1> *'), which are not related by inheritance, is not allowed
    return static_cast<_Base2*>(__pair);

Which the compiler narrowed down to `std::make_shared`.
